include/ck/utility/math_v2.hpp

```diff
@@ -3,63 +3,113 @@
 
 #include <cmath>
 #include "data_type.hpp"
-#include "half.hpp"
+#include "type.hpp"
 
 namespace ck {
 namespace math {
 
-static inline __host__ float abs(float x) { return std::abs(x); };
+// math functions for the host,  some are implemented by calling C++ std functions
 
-static inline __host__ double abs(double x) { return std::abs(x); };
+static inline __host__ float ck_abs(float x) { return std::abs(x); };
 
-static inline __host__ int8_t abs(int8_t x)
+static inline __host__ double ck_abs(double x) { return std::abs(x); };
+
+static inline __host__ int8_t ck_abs(int8_t x)
 {
     int8_t sgn = x >> (8 - 1);
 
     return (x ^ sgn) - sgn;
 };
 
-static inline __host__ int32_t abs(int32_t x)
+static inline __host__ int32_t ck_abs(int32_t x)
 {
     int32_t sgn = x >> (32 - 1);
 
     return (x ^ sgn) - sgn;
 };
 
-static inline __host__ half_t abs(half_t x)
+static inline __host__ half_t ck_abs(half_t x)
```

Comment:
Just want to make sure: it this implementation of ck_abs correct?

Reply:
Yes, for half_t the ck_abs() and ck_isnan() are implemented referring the codes in half.hpp. And it can be verified by checking the document describing the fp16 format

https://en.wikipedia.org/wiki/Half-precision_floating-point_format